tests/topology_test.cpp

```diff
@@ -120,7 +114,43 @@ TEST_CASE("get generation id") {
     REQUIRE(tc->get_generation_id() == 0);
     auto config = load_config("source/http");
     tc->add_config(config);
-    wait(1000);
     REQUIRE(tc->get_generation_id() == 1);
   });
+}
+
+TEST_CASE("test new topology") {
+  auto config = load_config("batch_file_to_file");
+  vectorcxx::test::setup_data();
+  REQUIRE(run_topology(config));
```

Comment:
Instead of introducing new API surfaces, is it possible to make this API part of `TopologyController`?

Reply:
For  example, `topology_controller->one_shot_run(config)`? so that users will always look for how to use `TopologyController` to do their jobs.